components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -74,49 +74,57 @@ preBuild.dependsOn(createBuildDir)
+            } else {
+                commandLine "$buildDir/bootstrap/Miniconda3/bin/python"
+            }
 
-        args "-c"
-        args runPythonScript
-        args "glean_parser"
-        args GLEAN_PARSER_VERSION
-        args "translate"
-        args "-f"
-        args "kotlin"
-        args "-o"
-        args "$sourceOutputDir"
-        args "$projectDir/metrics.yaml"
-
-        // Only show the output if something went wrong.
-        ignoreExitValue = true
-        standardOutput = new ByteArrayOutputStream()
-        errorOutput = standardOutput
-        doLast {
-            if (execResult.exitValue != 0) {
-                throw new GradleException("Process '${commandLine}' finished with non-zero exit value ${execResult.exitValue}:\n\n${standardOutput.toString()}")
+            args "-c"
+            args runPythonScript
+            args "glean_parser"
+            args GLEAN_PARSER_VERSION
+            args "translate"
+            args "-f"
+            args "kotlin"
+            args "-o"
+            args "$sourceOutputDir"
+            args "-s"
+            args "namespace=$namespace"
+            args "$projectDir/metrics.yaml"
+
+            // Only show the output if something went wrong.
+            ignoreExitValue = true
+            standardOutput = new ByteArrayOutputStream()
+            errorOutput = standardOutput
+            doLast {
+                if (execResult.exitValue != 0) {
+                    throw new GradleException("Process '${commandLine}' finished with non-zero exit value ${execResult.exitValue}:\n\n${standardOutput.toString()}")
+                }
             }
         }
-    }
 
-    // This is an Android-Gradle plugin 3+-ism.  Culted from reading the source,
-    // searching for "registerJavaGeneratingTask", and finding
-    // https://github.com/GoogleCloudPlatform/endpoints-framework-gradle-plugin/commit/2f2b91476fb1c6647791e2c6fe531a47615a1e85.
-    // The added directory doesn't appear in the paths listed by the
-    // `sourceSets` task, for reasons unknown.
-    variant.registerJavaGeneratingTask(generateKotlinAPI, new File(sourceOutputDir))
+        // This is an Android-Gradle plugin 3+-ism.  Culted from reading the source,
+        // searching for "registerJavaGeneratingTask", and finding
+        // https://github.com/GoogleCloudPlatform/endpoints-framework-gradle-plugin/commit/2f2b91476fb1c6647791e2c6fe531a47615a1e85.
+        // The added directory doesn't appear in the paths listed by the
+        // `sourceSets` task, for reasons unknown.
+        variant.registerJavaGeneratingTask(generateKotlinAPI, new File(sourceOutputDir))
+    }
 }
 
 if (android.hasProperty('applicationVariants')) {
-    android.applicationVariants.all(ext.gleanGenerateMetricsAPI)
+    android.applicationVariants.all(ext.gleanGenerateMetricsAPI('GleanMetrics'))
+} else {
+    android.libraryVariants.all(ext.gleanGenerateMetricsAPI('mozilla.components.service.glean.metrics'))
```

Comment:
What if we're collecting from a library? Is there any better way to assert that this is `glean`?

Reply:
Good question. I'll have to go spelunking into gradle internals, I guess, though it might be enough to inspect `buildDir` which probably has `glean` near the end...

Reply:
Mh, you know what? Maybe this can be as easy as additionally checking if we have `GLEAN_PING_SCHEMA_URL` in `BuildConfig`. That should not be true for non glean, right?

Reply:
I couldn't get that to work, but it does seem you can do `project.name == 'service-glean'`, which I think should do the trick.

Reply:
Sounds good to me!